File: LeetCode/Python Solutions/Integer Break/integer.py
class Solution:
    def integerBreak(self, n: int) -> int:
        dp = [0] * (n + 1)
        dp[1] = 1
        for i in range(2, n + 1):
            for j in range(1, i):
                dp[i] = max(dp[i], max(j, dp[j]) * max(i - j, dp[i - j]))
        return dp[n]


# Checking in PyCharm/terminal:
if __name__ == '__main__':
    Instant = Solution()
    Solve = Instant.integerBreak(n = 10)
    # n = 2 -> 1
    # n = 10 -> 36
    print(Solve)

File: LeetCode/Java Solutions/Integer Break/integer.java
public class Solution {
    public int integerBreak(int n) {
        int[] dp = new int[n + 1];
        dp[1] = 1;
        for (int i = 2; i <= n; i++) {
            for (int j = 1; j < i; j++) {
                dp[i] = Math.max(dp[i], Math.max(j, dp[j]) * Math.max(i - j, dp[i - j]));
            }
        }
        return dp[n];
    }
    
    public static void main(String[] args) {
        Solution solution = new Solution();
        int result = solution.integerBreak(10);
        System.out.println(result);  // Expected output: 36
    }
}
